likecoin/js/admin/likecoin_metabox.js

```diff
@@ -56,35 +56,128 @@ async function onISCNCallback(event) {
   ISCNTextField.innerHTML = `<a rel="noopener" target="_blank" href="${url}">${status}</a>`;
 }
 
+async function onLikePayCallbackAndUploadArweave(event) {
+  if (event.origin !== 'https://like.co') { // For development, skip this line.
+    return;
+  }
+  try {
+    const { action, data } = JSON.parse(event.data);
+    if (action !== 'TX_SUBMITTED') return;
+    const { tx_hash: txHash, error, success } = data;
+    if (error || success === false) return;
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/upload`,
+      dataType: 'json',
+      contentType: 'application/json; charset=UTF-8',
+      data: JSON.stringify({ txHash }), // LIKEpay txHash
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    if (!res.data) {
+      throw new Error('UPLOAD_TO_ARWEAVE_ERROR');
+    }
+    if (res.data.arweaveId && res.data.ipfsHash) {
+      onSubmitToISCN();
```

Comment:
Should not automatically upload to ISCN since there is an ISCN button, which is a separated button below arweave.
Should just refresh arweave ID information in both back and frontend, and make user click the already exist ISCN button instead

Reply:
Refactored to skip this chunk of code.